src/mindlessgen/molecules/generate_molecule.py

```diff
@@ -371,6 +377,39 @@ def generate_random_coordinates(at: np.ndarray) -> tuple[np.ndarray, np.ndarray]
     return xyz, ati
 
 
+def contract_coordinates(
+    xyz: np.ndarray, ati: np.ndarray, scale_bondlength: float
+) -> np.ndarray:
+    """
+    Pull the atoms towards the origin.
+    """
+    # Initialize the old coordinates as an array of zeros
+    xyz_old: np.ndarray = np.zeros_like(xyz)
+    cycle = 0
+    # Break if the coordinates do not change
+    while not np.array_equal(xyz_old, xyz):
+        cycle += 1
+        if cycle > 100000:
```

Comment:
Is 10^6 an appropriate number? Are 1000 or similar not also sufficient? We should use (roughly) a value that is maybe factor 10 to the maximally used value in standard runs.

Reply:
I tested it a little bit and according to the criteria i would suggest 2500 cycles.

Reply:
If I calculate correctly, this should correspond already to a raw distance of 500 Angstrom (which is quite a lot 😄).

Reply:
In 3 different runs with 110-120 atoms i got the number of cycles 264, 174 and 182. I just suggested roughly 10 times the value. But sure with 1000 cycles (should correspond to 200 Angstroem) this should be more than fine aswell.

Reply:
Let's leave it at 2500, sounds good to me.